waku/v2/protocol/waku_filter/waku_filter.nim

```diff
@@ -48,30 +48,22 @@ proc notify*(filters: Filters, msg: WakuMessage, requestId: string = "") =
     # TODO: In case of no topics we should either trigger here for all messages,
     # or we should not allow such filter to exist in the first place.
     for contentFilter in filter.contentFilters:
-      if contentFilter.contentTopics.len > 0:
-        if msg.contentTopic in contentFilter.contentTopics:
+        if msg.contentTopic == contentFilter.contentTopic:
           filter.handler(msg)
           break
 
 proc unsubscribeFilters(subscribers: var seq[Subscriber], request: FilterRequest, peerId: PeerID) =
   # Flatten all unsubscribe topics into single seq
-  var unsubscribeTopics: seq[ContentTopic]
-  for cf in request.contentFilters:
-    unsubscribeTopics = unsubscribeTopics.concat(cf.contentTopics)
-  
+  let unsubscribeTopics : seq[ContentTopic] = request.contentFilters.mapIt(it.contentTopic)
```

Comment:
More idiomatic to do assignment and operation separately, alternatively get rid of the type